Sources/Apodini/WebService.swift

```diff
@@ -128,28 +125,24 @@ extension WebService {
         
         self.register(SemanticModelBuilder(app))
     }
-    
-    
-    /// The current version of the `WebService`
-    public var version: Version {
-        Version()
-    }
 }
 
 
 extension WebService {
     func register(_ modelBuilder: SemanticModelBuilder) {
         let visitor = SyntaxTreeVisitor(modelBuilder: modelBuilder)
-        self.visit(visitor)
+        self.accept(visitor)
         visitor.finishParsing()
     }
     
-    func visit(_ visitor: SyntaxTreeVisitor) {
+    func accept(_ visitor: SyntaxTreeVisitor) {
         metadata.collectMetadata(visitor)
 
-        visitor.addContext(APIVersionContextKey.self, value: version, scope: .environment)
+        let version = visitor.currentNode.peekValue(for: APIVersionContextKey.self)
         visitor.addContext(PathComponentContextKey.self, value: [version], scope: .environment)
```

Comment:
I don't like how the `peekValue(for:)` so "far away" from the `SyntaxTreeVisitor`.  I could imagine solving this on a `Blackboard` basis in a cleaner way, but that would require some work. I suggest to leave it as is right now, but I'll make a note to #329. We should think about having default `Configuration`s on a `WebService`. This way we could have a `Configuration` that manipulates the paths of all endpoints by prepending the `Version` element. I always thought it should be possible not to include the path should by adding a `Configuration`. However, before we can implement #329 and this use case, I think a more sophisticated for ordering the execution of `Configuration`s should be implemented than just doing so _in order_.

Reply:
As a short term solution: An idea was to move all that logic (beginning and including the `metadata.collect...`) into the SyntaxTreeVisitor vistit method. However then the PathComponentContextKey would be added **outside** of the WebService, which then to me fells a bit misplaced.

What I would like to see in the future, could be a pattern based solution. Make the "path prefix" configurable by the user, and add some sort of version "pattern" (this can be string but also type based) which can be inserted by the user (defining a sensible default). This could also allow prefixes like `"/someservice-v2/..."`.
The actual insertion of the version is then done when parsing all the PathComponentContextKeys.

Just some thoughts coming up. I agree that there needs to be a better solution